In [46]:
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import numpy as np

In [47]:
df=pd.read_csv('test.csv',parse_dates=True)
df.head()

Review        date     Location
0  I was very impressed with the resort.\n Great ...  2019/08/20    Sebastian
1  The rooms were nice the outside needs work als...  2019/08/20  Los Angeles
2  Great location! I have stayed at this hotel on...  2019/08/20      Georgia
3  The hotel was adequate for my stay. The strips...  2019/08/20          NaN
4  Great location, room was large and spacious. P...  2019/08/19  Palm Harbor

In [48]:
df.isnull().sum()

Review        55
date           0
Location    4737
dtype: int64

In [49]:
df.shape

(6448, 3)

In [50]:
df.Review.astype(object)

0       I was very impressed with the resort.\n Great ...
1       The rooms were nice the outside needs work als...
2       Great location! I have stayed at this hotel on...
3       The hotel was adequate for my stay. The strips...
4       Great location, room was large and spacious. P...
                              ...                        
6443    ·the staff were all so friendly. frank, the ch...
6444           ·the internet barely worked or didn't work
6445    ·the location was great.\n \n \n \n \n ·the be...
6446    ·we were early for check in and the staff was ...
6447    ·staff, free coffee, location and activities.\...
Name: Review, Length: 6448, dtype: object

In [69]:
df1=df.replace(np.nan,'',regex=True)

In [70]:
df1.isnull().sum()

Review      0
date        0
Location    0
dtype: int64

In [71]:
df1['positive']=True
for i in range(6448):
    str=df1.loc[i,'Review'].lower()
    df1.loc[i,'positive']= ('happy' in str) or ('great' in str) or ('adequate' in str) or ('nice' in str) or ('love' in str) or ('excellent' in str)

In [72]:
df1.head()

Review        date     Location  \
0  I was very impressed with the resort.\n Great ...  2019/08/20    Sebastian   
1  The rooms were nice the outside needs work als...  2019/08/20  Los Angeles   
2  Great location! I have stayed at this hotel on...  2019/08/20      Georgia   
3  The hotel was adequate for my stay. The strips...  2019/08/20                
4  Great location, room was large and spacious. P...  2019/08/19  Palm Harbor   

   positive  
0      True  
1      True  
2      True  
3      True  
4      True

In [67]:
df1.positive.value_counts()

False    3704
True     2744
Name: positive, dtype: int64

In [68]:
2744/6448

0.42555831265508687

In [74]:
df.isnull().sum()

Review        55
date           0
Location    4737
dtype: int64

In [77]:
 locount=df1.groupby('Location').Location.count()

In [90]:
locount.sort_values(ascending=False)

Location
                            4737
United States of America     116
New York                      34
San Jose                      19
California                    19
                            ... 
Ocala                          1
Oc                             1
Oakland,CA                     1
Oakland, ca                    1
NRW, Germany                   1
Name: Location, Length: 1083, dtype: int64

In [91]:
df1['negative']=True
for i in range(6448):
    str=df1.loc[i,'Review'].lower()
    df1.loc[i,'negative']= ('unhappy' in str) or ('poor' in str) or ('disappoint' in str) or ('negative' in str) or ('wors' in str) or ('awful' in str)

In [92]:
df1.negative.value_counts()

False    6032
True      416
Name: negative, dtype: int64

In [104]:
df1['language']=''
from langdetect import detect

    

In [113]:
for i in range(6448):   
    str=df1.loc[i,'Review']
    try:
        lang = detect(str)
    except:
        lang='null'
        print("This row throws and error:", i)
    df1.loc[i,'language']=lang

This row throws and error: 33
This row throws and error: 263
This row throws and error: 507
This row throws and error: 560
This row throws and error: 877
This row throws and error: 1034
This row throws and error: 1091
This row throws and error: 1114
This row throws and error: 1150
This row throws and error: 1217
This row throws and error: 1244
This row throws and error: 1300
This row throws and error: 1355
This row throws and error: 1361
This row throws and error: 1385
This row throws and error: 1401
This row throws and error: 1479
This row throws and error: 1561
This row throws and error: 1625
This row throws and error: 1898
This row throws and error: 2119
This row throws and error: 2291
This row throws and error: 2365
This row throws and error: 2366
This row throws and error: 2391
This row throws and error: 2533
This row throws and error: 2610
This row throws and error: 2654
This row throws and error: 2663
This row throws and error: 2670
This row throws and error: 2671
This row throw

In [114]:
df1.head()

Review        date     Location  \
0  I was very impressed with the resort.\n Great ...  2019/08/20    Sebastian   
1  The rooms were nice the outside needs work als...  2019/08/20  Los Angeles   
2  Great location! I have stayed at this hotel on...  2019/08/20      Georgia   
3  The hotel was adequate for my stay. The strips...  2019/08/20                
4  Great location, room was large and spacious. P...  2019/08/19  Palm Harbor   

   positive  negative language  
0      True     False       en  
1      True     False       en  
2      True     False       en  
3      True      True       en  
4      True     False       en

In [116]:
df1.language.value_counts()

en       5864
es        167
de         88
fr         74
null       56
ja         36
it         35
pt         31
nl         24
da         18
no         10
ko          9
sv          8
et          6
af          5
zh-cn       4
fi          4
ca          2
pl          2
ru          2
he          1
zh-tw       1
ar          1
Name: language, dtype: int64

In [120]:
74/6448

0.011476426799007445

In [135]:
dat=df1.groupby('date').date.count()
dat=dat.sort_values(ascending=False)


In [136]:
pd.set_option("display.max_rows", None, "display.max_columns", None)
print(dat)

date
Jun 2019      145
May 2019      121
Jul 2019       92
2018/12/01     66
Oct 2018       58
2019/03/01     53
Aug 2018       53
2019/02/01     52
2018/09/01     49
2019/01/01     44
Nov 2018       42
2018/11/26     41
2018/09/04     38
2018/12/17     38
2019/07/31     37
Sep 2018       35
Mar 2019       35
2019/03/25     34
2018/10/01     33
2019/04/22     32
2019/04/01     32
Feb 2019       32
2019/01/02     31
2019/08/05     31
2018/11/01     31
2018/11/27     30
2018/09/10     29
2018/12/30     29
2019/03/26     28
2019/04/08     28
2019/1/4       28
2019/04/15     27
2018/12/31     27
2019/04/16     25
2019/04/21     25
2019/03/18     25
2018/12/18     25
2019/04/09     25
2018/10/02     25
2018/11/13     24
2018/10/15     24
2019/03/19     24
2019/1/15      24
2019/03/12     24
2019/05/20     24
2019/03/05     24
2019/08/09     24
2018/12/11     23
2018/10/23     23
2019/1/14      23
2019/04/23     23
2018/08/13     23
2019/08/11     23
2018/11/18     23
2018/11/19     23
2018/

1. There were 2744  clearly positive reviews among the total reviews. These reviews had positive keywords in the text  like 'happy', 'great', 'adequate', etc (only applies to the reviews in English). This means about 42.5% of the total reviews were positive.

2. 416 reviews had negative keywords in them, such as 'unhappy', 'poor', 'disappointed', etc.

3. 4737 entries had no location mentioned, and 55 entries had the review field blank.

4. Apart from null entries for 'Location', it was found that 'United States of America'  was the most popular location, with 116 entries, followed by New York  with 34, and 'San Jose' and 'California' with 19 each.

5. The reviews were written in multiple languages. A vast majority of these were written in English (5864/90.9%), followed by Spanish (167/2.5%), German (183/1.3%) and French (74/1.1%). There were also very few cases of Japanese, Korean and Arabic reviews. (This analysis was done using special libraries in python.)

6. 2018/12/01 saw the highest single-day traffic, i.e, it had the most number of reviews written in a single day, with 66 reviews. 2019/03/01 was the second-highest, with about 53 reviews.

All of the above insights were generated with the help of various libraries in python.
